modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CreateZoneCommand.java

```diff
@@ -0,0 +1,246 @@
+import org.apache.ignite.internal.catalog.DistributionZoneExistsValidationException;
+import org.apache.ignite.internal.catalog.descriptors.CatalogZoneDescriptor;
+import org.apache.ignite.internal.catalog.storage.NewZoneEntry;
+import org.apache.ignite.internal.catalog.storage.ObjectIdGenUpdateEntry;
+import org.apache.ignite.internal.catalog.storage.UpdateEntry;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A command that creates a new zone.
+ */
+public class CreateZoneCommand extends AbstractZoneCommand {
+    /** Returns builder to create a command to create a zone with specified name. */
+    public static CreateZoneCommandBuilder builder() {
+        return new Builder();
+    }
+
+    private final @Nullable Integer partitions;
+
+    private final @Nullable Integer replicas;
+
+    private final @Nullable Integer dataNodesAutoAdjust;
+
+    private final @Nullable Integer dataNodesAutoAdjustScaleUp;
+
+    private final @Nullable Integer dataNodesAutoAdjustScaleDown;
+
+    private final @Nullable String filter;
+
+    private final @Nullable DataStorageParams dataStorageParams;
+
+    /**
+     * Constructor.
+     *
+     * @param zoneName Name of the zone.
+     * @param partitions Number of partitions.
+     * @param replicas Number of replicas.
+     * @param dataNodesAutoAdjust Timeout in seconds between node added or node left topology event itself and data nodes switch.
+     * @param dataNodesAutoAdjustScaleUp Timeout in seconds between node added topology event itself and data nodes switch.
+     * @param dataNodesAutoAdjustScaleDown Timeout in seconds between node left topology event itself and data nodes switch.
+     * @param filter Nodes filter.
+     * @param dataStorageParams Data storage params.
+     * @throws CatalogValidationException if any of restrictions above is violated.
+     */
+    private CreateZoneCommand(
+            String zoneName,
+            @Nullable Integer partitions,
+            @Nullable Integer replicas,
+            @Nullable Integer dataNodesAutoAdjust,
+            @Nullable Integer dataNodesAutoAdjustScaleUp,
+            @Nullable Integer dataNodesAutoAdjustScaleDown,
+            @Nullable String filter,
+            @Nullable DataStorageParams dataStorageParams
+    ) throws CatalogValidationException {
+        super(zoneName);
+
+        this.partitions = partitions;
+        this.replicas = replicas;
+        this.dataNodesAutoAdjust = dataNodesAutoAdjust;
+        this.dataNodesAutoAdjustScaleUp = dataNodesAutoAdjustScaleUp;
+        this.dataNodesAutoAdjustScaleDown = dataNodesAutoAdjustScaleDown;
+        this.filter = filter;
+        this.dataStorageParams = dataStorageParams;
+
+        validate();
+    }
+
+    @Override
+    public List<UpdateEntry> get(Catalog catalog) {
+        if (catalog.zone(zoneName) != null) {
+            throw new DistributionZoneExistsValidationException("Distribution zone already exists [zoneName=" + zoneName + ']');
```

Comment:
The patter for this type of exception is `<Object> with name '{}' already exists` (see `org.apache.ignite.internal.catalog.CatalogParamsValidationUtils#ensureNoTableIndexOrSysViewExistsWithGivenName`)